documentation/src/main/asciidoc/mapper-orm-indexing-explicit.asciidoc

```diff
@@ -0,0 +1,374 @@
+Note we're relying on automatic indexing to index the entity,
+but this would work just as well if automatic indexing was disabled,
+only requiring an extra call to index the entity.
+See <<mapper-orm-indexing-explicit-writeplan-writes>>.
+<5> Commit the transaction at the end of each iteration of the outer loop.
+The entities will be flushed and indexed automatically.
+====
+
+Finally, if the other solutions are unacceptable because
+the index needs to stay in its original state when the batch process fails,
+or because the process cannot be broken down in small enough transactions,
+there is a third solution: just tell Hibernate Search to process the entities
+after the call to `session.flush()` and before the call to `session.clear()`,
+without waiting for the database transaction to be committed.
+Hibernate Search will read the entities,
+build the corresponding documents and store them in an internal buffer,
+which will be written to indexes upon commit.
+
+This is done by calling the `process()` method on the write plan,
+as shown in the example below.
+
+[IMPORTANT]
+====
+This pattern will require less memory than without the periodic `flush`/`process`/`clear`,
+but it will still maintain an internal buffer that will grow as more entities are processed,
+and may lead to an `OutOfMemoryException` depending on the JVM heap size and the complexity and number of documents.
+Only use this in production after careful testing with an appropriate JVM heap size.
+
+However, if enough memory is available,
+this is the safest pattern, leaving both the database and indexes unaffected
+if an error happens during the batch process.
+====
+
+.A batch process with Hibernate Search using `process()`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/indexing/HibernateOrmExplicitIndexingIT.java[tags=persist-automatic-indexing-periodic-process]
+----
+<1> Get the `SearchSession`.
+<2> Get the search session's write plan.
+<3> For every iteration of the loop, instantiate a new entity and persist it.
+Note we're relying on automatic indexing to index the entity,
+but this would work just as well if automatic indexing was disabled,
+only requiring an extra call to index the entity.
+See <<mapper-orm-indexing-explicit-writeplan-writes>>.
+<4> After after a `flush()`, call `writePlan.process()`.
+The entities will be processed but *the changes will not be sent to the indexes*,
+and will instead be stored in an internal buffer.
+<5> After the loop, commit the transaction.
+The changes stored in the internal buffer, as well as the remaining entities that were not flushed/cleared,
+will be indexed automatically.
+====
+
+[NOTE]
+====
+The multi-transaction solution and the periodic `process()` solution can be combined,
+breaking down the process in multiple medium-sized transactions,
+and periodically calling `flush`/`process`/`clear` inside each transaction.
```

Comment:
Maybe we can add that this last combined strategy is the more flexible and the more suitable for a fine tuning?